op-chain-ops/cmd/withdrawals/main.go

```diff
@@ -33,60 +165,518 @@ func main() {
+			log.Debug("Migration data", "ovm-path", ovmMsgs, "evm-messages", evmMsgs)
+			ovmMessages, err := migration.NewSentMessage(ovmMsgs)
+			if err != nil {
+				return err
+			}
+			evmMessages, err := migration.NewSentMessage(evmMsgs)
 			if err != nil {
 				return err
 			}
 
-			start := ctx.Uint64("start")
-			end := ctx.Uint64("end")
+			optimismPortalAddress := ctx.String("optimism-portal-address")
+			if len(optimismPortalAddress) == 0 {
+				return errors.New("OptimismPortal address not configured")
+			}
+			optimismPortalAddr := common.HexToAddress(optimismPortalAddress)
 
-			// All messages are expected to be version 0 messages
-			withdrawals, err := crossdomain.GetPendingWithdrawals(messengers, common.Big0, start, end)
+			migrationData := migration.MigrationData{
+				OvmMessages: ovmMessages,
+				EvmMessages: evmMessages,
+			}
+
+			// create the set of withdrawals
+			wds, err := migrationData.ToWithdrawals()
 			if err != nil {
 				return err
 			}
+			if len(wds) == 0 {
+				return errors.New("no withdrawals")
+			}
+			log.Info("Converted migration data to withdrawals successfully", "count", len(wds))
+
+			l1xdmAddress := ctx.String("l1-crossdomain-messenger-address")
+			if l1xdmAddress == "" {
+				return errors.New("Must pass in --l1-crossdomain-messenger-address")
+			}
+			l1xdmAddr := common.HexToAddress(l1xdmAddress)
+
+			l1CrossDomainMessenger, err := bindings.NewL1CrossDomainMessenger(l1xdmAddr, l1Client)
+			if err != nil {
+				return err
+			}
+
+			portal, err := bindings.NewOptimismPortal(optimismPortalAddr, l1Client)
+			if err != nil {
+				return err
+			}
+
+			l2OracleAddr, err := portal.L2ORACLE(&bind.CallOpts{})
+			if err != nil {
+				return err
+			}
+			oracle, err := bindings.NewL2OutputOracle(l2OracleAddr, l1Client)
+			if err != nil {
+				return nil
+			}
+
+			log.Info(
+				"Addresses",
+				"l1-crossdomain-messenger", l1xdmAddr,
+				"optimism-portal", optimismPortalAddr,
+				"output-oracle", l2OracleAddr,
+			)
+
+			period, err := portal.FINALIZATIONPERIODSECONDS(&bind.CallOpts{})
+			if err != nil {
+				return err
+			}
+
+			transitionBlockNumber := new(big.Int).SetUint64(ctx.Uint64("bedrock-transition-block-number"))
+			log.Info("Withdrawal config", "finalization-period", period, "bedrock-transition-block-number", transitionBlockNumber)
+
+			if ctx.String("private-key") == "" {
+				return errors.New("No private key to transact with")
+			}
+			privateKey, err := crypto.HexToECDSA(strings.TrimPrefix(ctx.String("private-key"), "0x"))
+			if err != nil {
+				return err
+			}
+
+			type badWithdrawal struct {
+				Withdrawal *crossdomain.Withdrawal       `json:"withdrawal"`
+				Legacy     *crossdomain.LegacyWithdrawal `json:"legacy"`
+				Trace      callFrame                     `json:"trace"`
+				Index      int                           `json:"index"`
+			}
+
+			badWithdrawals := make([]badWithdrawal, 0)
+
+			// iterate over all of the withdrawals and submit them
```

Comment:
This is more script like, but even for a script, this is a very long function.

Reply:
You are right and this was bothering me so i refactored it a bunch